runtime/integration-tests/pendulum/src/tests.rs

```diff
@@ -0,0 +1,350 @@
+use crate::{polkadot_test_net::*, setup::*};
+use frame_support::{
+	assert_ok,
+	traits::{fungible::Mutate, fungibles::Inspect, Currency},
+};
+use pendulum_runtime::{Balances, PendulumCurrencyId, RuntimeOrigin, Tokens, XTokens};
+use sp_runtime::{traits::AccountIdConversion, MultiAddress};
+use xcm::latest::{Junction, Junction::*, Junctions::*, MultiLocation, NetworkId, WeightLimit};
+use xcm_emulator::{Junctions, TestExt};
+
+use polkadot_core_primitives::{AccountId, Balance};
+use polkadot_parachain::primitives::Sibling;
+
+const DOT_FEE: Balance = 3200000000;
+const ASSET_ID: u32 = 1984; //Real USDT Asset ID from Statemint
+const INCORRECT_ASSET_ID: u32 = 0;
+pub const UNIT: Balance = 1_000_000_000_000;
+pub const TEN: Balance = 10_000_000_000_000;
+const FEE: u128 = 421434140;
+
+#[test]
+fn transfer_polkadot_from_relay_chain_to_pendulum() {
+	MockNet::reset();
+
+	let transfer_amount: Balance = dot(20);
+	let mut orml_tokens_before = 0;
+	PendulumParachain::execute_with(|| {
+		orml_tokens_before = pendulum_runtime::Tokens::balance(
+			pendulum_runtime::PendulumCurrencyId::XCM(0),
+			&ALICE.into(),
+		);
+	});
+
+	Relay::execute_with(|| {
+		assert_ok!(polkadot_runtime::XcmPallet::reserve_transfer_assets(
+			polkadot_runtime::RuntimeOrigin::signed(ALICE.into()),
+			Box::new(Parachain(2094).into().into()),
+			Box::new(Junction::AccountId32 { network: NetworkId::Any, id: ALICE }.into().into()),
+			Box::new((Here, transfer_amount).into()),
+			0
+		));
+	});
+
+	PendulumParachain::execute_with(|| {
+		use pendulum_runtime::{RuntimeEvent, System};
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::Tokens(orml_tokens::Event::Deposited { .. })
+		)));
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::DmpQueue(cumulus_pallet_dmp_queue::Event::ExecutedDownward { .. })
+		)));
+	});
+
+	PendulumParachain::execute_with(|| {
+		assert_eq!(
+			pendulum_runtime::Tokens::balance(
+				pendulum_runtime::PendulumCurrencyId::XCM(0),
+				&ALICE.into()
+			),
+			orml_tokens_before + transfer_amount - DOT_FEE
+		);
+	});
+}
+
+#[test]
+fn transfer_polkadot_from_pendulum_to_relay_chain() {
+	MockNet::reset();
+
+	let transfer_dot_amount: Balance = dot(10);
+
+	Relay::execute_with(|| {
+		let after_bob_free_balance = polkadot_runtime::Balances::free_balance(&BOB.into());
+		assert_eq!(after_bob_free_balance, dot(100));
```

Comment:
This should be called `before_bob_free_balance` probably.

Reply:
Maybe we can extract the `dot(100)` to an extra variable and put it in line 74. as
```
let expected_base_balance = dot(100)
```

and then you can also replace the `dot(100)` in the assert statement in l. 131 with that variable.